src/rpc/mining.cpp

```diff
@@ -70,75 +70,97 @@ static UniValue GetNetworkHashPS(int lookup, int height, std::string type) {
     if (pb == nullptr || !pb->nHeight)
         return 0;
 
+    bool isPoS = type == "pos";
+    bool isSha256D = type == "sha256d";
+    bool isRandomX = type == "randomx";
+    bool isProgPow = type == "progpow";
+    bool isX16RT = type == "xr16t";
```

Comment:
Nit: this could probably be turned into an enum value and then used as a switch in the if-else blocks below.

Reply:
True, that's the cleaner way, just changed that.